src/client.ts

```diff
@@ -58,63 +52,47 @@ export class APIClient {
    *
    * @param iteratee receives each resource to produce entities/relationships
    */
-  public async iterateUsers(
-    iteratee: ResourceIteratee<AcmeUser>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const users: AcmeUser[] = [
-      {
-        id: 'acme-user-1',
-        name: 'User One',
-      },
-      {
-        id: 'acme-user-2',
-        name: 'User Two',
-      },
-    ];
-
-    for (const user of users) {
-      await iteratee(user);
+  public async iterateUsers(iteratee: ResourceIteratee<User>): Promise<void> {
+    const apiInstance = new v1.UsersApi(this.configuration);
+    const res = await apiInstance.listUsers();
+    if (res.users) {
+      for (const user of res.users) {
+        await iteratee(user);
+      }
+    } else {
+      return;
     }
   }
 
   /**
-   * Iterates each group resource in the provider.
+   * Fetches the singleton top-level account resource in the provider.
+   */
+  public async getAccount(): Promise<User | undefined> {
+    const apiInstance = new v1.UsersApi(this.configuration);
+    const account = await apiInstance.getUser({
+      userHandle: this.config.datadogAccountEmail,
+    });
+    return account.user;
+  }
+
+  /**
+   * Iterates each role resource in the provider.
    *
    * @param iteratee receives each resource to produce entities/relationships
    */
-  public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
-          {
-            id: 'acme-user-1',
-          },
-        ],
+  public async iterateRoles(iteratee: ResourceIteratee<Role>): Promise<void> {
+    const configuration = v2.createConfiguration({
+      authMethods: {
+        apiKeyAuth: this.config.datadogApiKey,
+        appKeyAuth: this.config.datadogAppKey,
```

Comment:
Had to use the v2 API client specifically for roles because the v1 client does not support it. Was thinking of using v2 for all resources but v2 does not some endpoints too. So I guess we are going to have to use a combination of both for now.